src/ui/server/controllers/insurance/policy-and-export/about-goods-or-services/index.ts

```diff
@@ -65,3 +68,51 @@ export const get = (req: Request, res: Response) => {
     application,
   });
 };
+
+/**
+ * post
+ * Check About goods or services validation errors and if successful, redirect to the next part of the flow.
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} Next part of the flow or error page
+ */
+export const post = async (req: Request, res: Response) => {
+  const { application } = res.locals;
+
+  if (!application) {
+    return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+  }
+
+  const { referenceNumber } = req.params;
+  const refNumber = Number(referenceNumber);
+
+  const validationErrors = generateValidationErrors(req.body);
+
+  if (validationErrors) {
+    return res.render(TEMPLATE, {
+      ...insuranceCorePageVariables({
+        PAGE_CONTENT_STRINGS: PAGES.INSURANCE.POLICY_AND_EXPORTS.ABOUT_GOODS_OR_SERVICES,
+        BACK_LINK: req.headers.referer,
+      }),
+      ...pageVariables(refNumber),
+      application,
+      submittedValues: req.body,
+      validationErrors,
+    });
+  }
+
+  try {
+    // save the application
+    const saveResponse = await mapAndSave.policyAndExport(req.body, application);
+
+    if (!saveResponse) {
+      return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+    }
+
+    return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${ALL_SECTIONS}`);
+  } catch (err) {
+    console.error('Error updating application', { err });
```

Comment:
is it worth adding which section we were in to the console.error so it's easier to track where it failed if looking at logs?